Meeting-15/README.md

```diff
@@ -1,28 +1,286 @@
+
+```javascript
+import { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+const navigate = useNavigate();
+
+useEffect(() => {
+  // check if user is admin
+  const isAdmin = false;
 
-[URL Parameters :ReactRouter.com](https://reactrouter.com/web/example/url-params)
+  // if not redirect the user to /
+  if (!isAdmin) {
+    navigate("/");
+  }
+}, []);
+```
+
+**Takeaways**
+
+-The useNavigate() hook allows us to programmatically redirect the user.
+
+- useNavigate() will return us the navigate function where we can use it whenever we want to redirect the user.
+
+## Grouping Routes
+
+As we are building our application it is unavoidable that we would have routes that has same prefix. For example let's say we have several admin routes:
+
+```javascript
+<Routes>
+	<Route path="/admin/dashboard" element={<Dashboard />}>
+	<Route path="/admin/manage" element={<Dashboard />}>
+	<Route path="/admin/settings" element={<Dashboard />}>
+</Routes>
+```
+
+Now all of this routes has same prefix called 'admin', however if we write our route component just like what we did here we essentially are repeating ourselves. React router has a solution which is called Route Grouping. Rather then declaring routes one by one what we can do instead is to have a parent route component where we define the admin path prefix on it and within the route component we'll define the individual children routes.
+
+```javascript
+<Routes>
+	<Route path="/admin">
+		<Route path="dashboard" element={<Dashboard />}>
+		<Route path="manage" element={<Dashboard />}>
+		<Route path="settings" element={<Dashboard />}>
+	</Route>
+</Routes>
+```
+
+One important note here: we can not put a slash in front of our path, because a slash means an absolute path in React Router. If we put a slash in front of a path it means when a user visits the slash dashboard endpoint we would like to serve the user the dashboard component and that defeats the purpose of Route grouping all together.
+
+**Takeaways**
+
+-We can put additional <Route> components within another to group multiple routes together.
+
+## Dynamic routes
+
+In a situation where we need to use dynamic routes. For example in an admin page where we need to manage a spacific user. The routes will look somethin like this: 'admin/users/1'.
+
+```javascript
+<Routes>
+	<Route path="/admin">
+		<Route path="dashboard" element={<Dashboard />}>
+		<Route path="manage" element={<Dashboard />}>
+		<Route path="settings" element={<Dashboard />}>
+		<Route path="users/:userId" element={<Dashboard />}>
+	</Route>
+</Routes>
+```
+
+where ':userId' is a dynamic parameter.
+
+To retrieve a dynamic URL parameter we'll need to use the useParams hook. It will return us an object that represents all the URL parameters in the current React route. The value is whatever we put inside a URL and the key refers to the name of the parameter what is registered within a route component (in this example 'userId').
```

Comment:
```suggestion
To retrieve a dynamic URL parameter we'll need to use the `useParams` hook. It will return us an object that represents all the URL parameters in the current React route.

The **value** is whatever we put inside a URL and the **key** refers to the name of the parameter that is registered within a route component (in this example `'userId'`).
```